### ISSUE TITLE
Low Performance on FashionIQ

### ISSUE THREAD

**Issue by hockeybro12** [ecom-research/ComposeAE#21]
Hello,

I trained a model on FashionIQ (as described in the ReadME), but I am getting low performance even after running it for 300 epochs. I trained the models like so: ```python -W ignore  main.py --dataset=fashionIQ --dataset_path=data/fashionIQ/  --model=RealSpaceConcatAE --loss=batch_based_classification --learning_rate_decay_frequency=8000 --num_iters=100000 --use_bert True --use_complete_text_query True --comment=fashionIQ_RealSpaceConcatAE --log_dir logs/fashionIQ/ --batch_size 128```

Here are my Tensorboard logs. You can see the loss is decreasing:

<img width="396" alt="Screen Shot 2021-07-11 at 10 48 26 AM" src="https://user-images.githubusercontent.com/5779609/125205109-87895300-e235-11eb-8900-5519a64a2498.png">

But the performance on the train/test set in terms of recall is bad. What could be the issue?
<img width="426" alt="Screen Shot 2021-07-11 at 10 49 23 AM" src="https://user-images.githubusercontent.com/5779609/125205132-aab40280-e235-11eb-8bdd-76c5414eecbd.png">

<img width="413" alt="Screen Shot 2021-07-11 at 10 49 37 AM" src="https://user-images.githubusercontent.com/5779609/125205143-b1db1080-e235-11eb-91ae-3876885cea7f.png">

<img width="402" alt="Screen Shot 2021-07-11 at 10 49 54 AM" src="https://user-images.githubusercontent.com/5779609/125205154-bb647880-e235-11eb-900a-ba32d2c377c7.png">

**Comment by ecom-research** [comment]
First thing to note is that the results on Fashion-IQ dataset are on the validation set of the dataset. The test set was not revealed to the public by the curators of the dataset at the time of writing of the paper. I am not sure if they have already published the test set or not.
Secondly, I saw that you ran "RealSpaceConcatAE".  Can you please also run "ComposeAE" model and see what results you get there?

**Comment by ecom-research** [comment]
Thirdly, I see in your last image that you have test/validation recall still rising. Maybe keep on training., 

Fourthly, is the pre-trained ResNet model kept trainable?

**Comment by ecom-research** [comment]
Lastly, I will repeat again how we used the dataset here. Kindly follow these:

"FashionIQ dataset
Download the dataset via this [link](https://github.com/XiaoxiaoGuo/fashion-iq) and save it in the data folder. The dataset consists of three non-overlapping subsets, namely dress, top-tee and shirt. We join the two annotations with the text and it to get a description similar to a normal sentence a user might ask on an E-Com platform. Furthermore, we combine the train sets of all three categories to form a bigger training set and train a single model on it. Analogously, we also combine the validation sets to form a single validation set."

**Comment by hockeybro12** [comment]
```First thing to note is that the results on Fashion-IQ dataset are on the validation set of the dataset. The test set was not revealed to the public by the curators of the dataset at the time of writing of the paper. I am not sure if they have already published the test set or not.```

Yes, I am running on the validation set. In fact, your code already does that by default, as you can see here: https://github.com/ecom-research/ComposeAE/blob/271865ecad1e0efb101a6a2be4915ea7dd263e61/datasets.py#L404

```Secondly, I saw that you ran "RealSpaceConcatAE". Can you please also run "ComposeAE" model and see what results you get there?```

I ran that and while the results are slightly better, they aren't near high enough still.

```Thirdly, I see in your last image that you have test/validation recall still rising. Maybe keep on training.,```
Do you remember how many iterations you ran? So far I haven't been able to get the performance

```Fourthly, is the pre-trained ResNet model kept trainable?```
I believe it should be: https://github.com/ecom-research/ComposeAE/blob/271865ecad1e0efb101a6a2be4915ea7dd263e61/img_text_composition_models.py#L120


```Lastly, I will repeat again how we used the dataset here. Kindly follow these:

"FashionIQ dataset
Download the dataset via this link and save it in the data folder. The dataset consists of three non-overlapping subsets, namely dress, top-tee and shirt. We join the two annotations with the text and it to get a description similar to a normal sentence a user might ask on an E-Com platform. Furthermore, we combine the train sets of all three categories to form a bigger training set and train a single model on it. Analogously, we also combine the validation sets to form a single validation set."```

Again, it seems like your code already does this and I can verify manually that the dataset class does in fact do this.

Joining the two annotations: https://github.com/ecom-research/ComposeAE/blob/271865ecad1e0efb101a6a2be4915ea7dd263e61/datasets.py#L444

Using the whole training set: https://github.com/ecom-research/ComposeAE/blob/271865ecad1e0efb101a6a2be4915ea7dd263e61/datasets.py#L438

Do you have any other ideas for what may be wrong?

**Comment by ecom-research** [comment]
For all experiments, we ran it for 200k iterations and did a grid search for some hyperparams.
Unfortunately, I don't remember exactly what was the learning rate for Fashion IQ.
The default values for all hyperparameters in the code are set for MIT-States dataset. For other datasets, you have to do grid search. 
Due to high workload these days, I may not be able to help you further in pinning down the issues you are facing. I apologize in advance.

**Comment by hockeybro12** [comment]
> For all experiments, we ran it for 200k iterations and did a grid search for some hyperparams.
> Unfortunately, I don't remember exactly what was the learning rate for Fashion IQ.
> The default values for all hyperparameters in the code are set for MIT-States dataset. For other datasets, you have to do grid search.
> Due to high workload these days, I may not be able to help you further in pinning down the issues you are facing. I apologize in advance.

Hmm, ok. I will try different parameters then. It would also be very helpful if you could release the pre-trained model if you have it, as that could also help with reproducibility.

**Comment by ecom-research** [comment]
I have uploaded one model from our run of FashionIQ. Kindly access it here: [https://ltr-log-dataset.s3.eu-central-1.amazonaws.com/FashionIQ_Model_ComposeAE/FIQ_ComposeAE_checkpoint.pth](url)

I hope that you understand further requests will be difficult for us to work upon.
